arcgis-ios-sdk-samples/Maps/Create and save a map/CreateSaveMapViewController.swift

```diff
@@ -35,169 +35,124 @@ extension UIImage {
+    func createOptionsViewController(_ createOptionsViewController: CreateOptionsViewController, didSelectBasemap basemap: AGSBasemap, layers: [AGSLayer]) {
         
         //create a map with the selected basemap
         let map = AGSMap(basemap: basemap)
         
         //add the selected operational layers
-        if let layers = layers {
-            map.operationalLayers.addObjects(from: layers)
-        }
+        map.operationalLayers.addObjects(from: layers)
+        
         //assign the new map to the map view
-        self.mapView.map = map
+        mapView.map = map
         
-        //hide the create view
-        self.toggleCreateView()
+        createOptionsViewController.dismiss(animated: true)
     }
     
     // MARK: - SaveAsVCDelegate
     
     func saveAsViewController(_ saveAsViewController: SaveAsViewController, didInitiateSaveWithTitle title: String, tags: [String], itemDescription: String) {
+        
         SVProgressHUD.show(withStatus: "Saving")
+        
         //set the initial viewpoint from map view
-        self.mapView.map?.initialViewpoint = self.mapView.currentViewpoint(with: AGSViewpointType.centerAndScale)
+        mapView.map?.initialViewpoint = mapView.currentViewpoint(with: AGSViewpointType.centerAndScale)
         
-        self.mapView.exportImage { [weak self] (image: UIImage?, error: Error?) in
+        mapView.exportImage { [weak self] (image: UIImage?, error: Error?) in
+            
+            guard let self = self else {
+                return
+            }
             
-            if let weakSelf = self {
-                //crop the image from the center
-                //also to cut on the size
-                let croppedImage: UIImage? = image?.croppedImage(CGSize(width: 200, height: 200))
-                
-                weakSelf.mapView.map?.save(as: title, portal: weakSelf.portal!, tags: tags, folder: nil, itemDescription: itemDescription, thumbnail: croppedImage, forceSaveToSupportedVersion: true) { [weak self] (error) in
-                    //dismiss progress hud
-                    SVProgressHUD.dismiss()
-                    if let error = error {
-                        self?.presentAlert(error: error)
-                    } else {
+            //crop the image from the center
+            //also to cut on the size
+            let croppedImage: UIImage? = image?.croppedImage(CGSize(width: 200, height: 200))
+            
+            self.mapView.map?.save(as: title, portal: self.portal!, tags: tags, folder: nil, itemDescription: itemDescription, thumbnail: croppedImage, forceSaveToSupportedVersion: true) { [weak self] (error) in
+                //dismiss progress hud
+                SVProgressHUD.dismiss()
+                if let error = error {
+                    saveAsViewController.presentAlert(error: error)
+                    saveAsViewController.dismiss(animated: true)
```

Comment:
I'm confused by this. Why is the view controller being dismissed if it just presented an alert?

Reply:
Good point. I updated it so it's dismissed first and then the parent view controller presents the alert.

Reply:
> I updated it so it's dismissed first

I kind of like how the Save As view stayed up when there was an error. It made it easier to try again. That being said, it isn't a big deal.

Since both branches now dismiss the Save As view controller, you could combine the two calls to `dismiss` by moving the `if` statement into the completion handler:

```swift
saveAsViewController.dismiss(animated: true) {
    if let error = error {
        self?.presentAlert(error: error)
    } else {
        self?.showSuccess()
    }
}
```

Reply:
> I kind of like how the Save As view stayed up when there was an error. It made it easier to try again.

Ohh I didn't realize this was desired behavior. Your code from before makes more sense now.

Reply:
Okay, this should be good to go.